interview-questions.md

```diff
@@ -8,50 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: a parameter is something that we give a function so that it takes that parameter, and inputs it into the function with the user's input. An argument is whatever the user decides to input into said function, the argument is then passed into the function in place of the parameter. for example, a function that has an array parameter, can take a variable as an argument. The variable the user makes is can be an array that they decide to make. The argument is then passed as the parameter.
+
+Researched answer: parameters are named variables passed into functions. Parameters are used to import arguments into the functions. Function parameters are names that are listed in the definition of a function. Function arguments are then the actual values passed into the function. 
+Source: https://developer.mozilla.org/en-US/docs/Glossary/Parameter
+
+
+
 
-Researched answer:
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: Javascript built in method is a function. It can be used inside of other functions. Since the .map method is a function, it can take parameters. The parameters it can take are, value, index, and array. The only parameter that is required is value, because we must pass in a value argument in order for the function .map to do its thing. The optional parameters are index and array. 
+
+Researched answer: the .map method is a callback function. It is a copying method that can also mutate arrays. It requires a return since it is a function.
```

Comment:
This would be a great area to talk about why index and array parameters aren't required and/or how they could be used.